metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py

```diff
@@ -144,15 +139,22 @@ def bigquery_audit_metadata_query_template(
             AND timestamp < "{{end_time}}"
         )
         {shard_condition}
-        AND (
-                (
-                    protopayload_auditlog.serviceName="bigquery.googleapis.com"
-                    AND JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson, "$.jobChange.job.jobStatus.jobState") = "DONE"
-                    AND JSON_EXTRACT(protopayload_auditlog.metadataJson, "$.jobChange.job.jobConfig.queryConfig") IS NOT NULL
-                    {allow_filter}
-                )
+        AND protopayload_auditlog.serviceName="bigquery.googleapis.com"
+        AND
+        (
+            (
+                JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson, "$.jobChange.job.jobStatus.jobState") = "DONE"
+                AND JSON_EXTRACT(protopayload_auditlog.metadataJson, "$.jobChange.job.jobStatus.errorResults") IS NULL
+                AND JSON_EXTRACT(protopayload_auditlog.metadataJson, "$.jobChange.job.jobConfig.queryConfig") IS NOT NULL
+                AND (
+                        JSON_EXTRACT_ARRAY(protopayload_auditlog.metadataJson,
+                                                            "$.jobChange.job.jobStats.queryStats.referencedTables") IS NOT NULL
+                    OR
+                        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson, "$.jobChange.job.jobConfig.queryConfig.destinationTable") IS NOT NULL
+                    )
+            )
             OR
-            JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson, "$.tableDataRead.reason") = "JOB"
+                JSON_EXTRACT(protopayload_auditlog.metadataJson, "$.tableDataRead") IS NOT NULL
```

Comment:
Are there other reasons on https://cloud.google.com/bigquery/docs/reference/auditlogs/rest/Shared.Types/BigQueryAuditMetadata.TableDataRead.Reason that we want to ingest? Specifically `MATERIALIZED_VIEW_REFRESH` seems like something we want to skip. The rest I'm not sure about.

Also, what is this OR block actually doing? How come we're okay skipping all the above conditionals if tableDataRead exists?

Reply:
This change was done, to keep this filter in sync with [GCP logging filters](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_audit.py#L42)(i.e. when not using exported audit logs). 


As to which reasons do we want to ingest - that's a good question to think about.

> what is this OR block actually doing? How come we're okay skipping all the above conditionals if tableDataRead exists

The audit log entry's [BigqueryAuditMetadata](https://cloud.google.com/bigquery/docs/reference/auditlogs/rest/Shared.Types/BigQueryAuditMetadata#BigQueryAuditMetadata.TableDataRead) can either contain `tableDataRead` or `jobChange`. This OR is to consider both kind of metadata entries.

Reply:
Probably we can consider only "JOB" type to start with ?

Reply:
I'm guessing `AND` takes precedence over `OR` but we prob good to put parentheses around these 2 to be safe. Or maybe we can just remove `JSON_EXTRACT(protopayload_auditlog.metadataJson, "$.tableDataRead") IS NOT NULL`, if the check still works without it?